### ISSUE TITLE
An internal error occurred while type checking file TypeError: Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by jdogmcsteezy** [microsoft/pyright#1569]
**Describe the bug**
From VSCode:
```
An internal error occurred while type checking file "/mnt/hgfs/user/Developer/aerospike-admin/asadm.py": TypeError: Cannot read property 'id' of undefined
    at findIndex (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:802:50)
    at Array.findIndex (<anonymous>)
    at getIndexOfSymbolResolution (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:801:38)
    at setSymbolResolutionPartialType (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:831:23)
    at getTypeOfFunction (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:11449:13)
    at getTypeForDeclaration (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16290:42)
    at getDeclaredTypeOfSymbol (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16673:38)
    at getEffectiveTypeOfSymbolForUsageInternal (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16527:34)
    at callback (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16515:19)
    at X (/home/user/.vscode-server/extensions/ms-python.vscode-pylance-2021.2.4/dist/pyright-internal/src/analyzer/typeEvaluator.ts:697:15)
```

From Pyright CLI:
```
> pyright
No configuration file found.
stubPath /mnt/hgfs/jesseschmidt/Developer/public-aerospike/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 97 source files
An internal error occurred while type checking file "/mnt/hgfs/user/Developer/public-aerospike/asadm.py": TypeError: Cannot read property 'id' of undefined
    at /usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:310999
    at Array.findIndex (<anonymous>)
    at re (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:310970)
    at le (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:311299)
    at Rt (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:429306)
    at Pn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:482222)
    at On (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:486024)
    at cn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:458095)
    at dn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:459191)
    at e (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:465298)
Error performing analysis: TypeError: Cannot read property 'id' of undefined
    at /usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:310999
    at Array.findIndex (<anonymous>)
    at re (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:310970)
    at le (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:311299)
    at Rt (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:429306)
    at Pn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:482222)
    at On (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:486024)
    at cn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:458095)
    at dn (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:459191)
    at e (/usr/local/lib/node_modules/pyright/dist/pyright-internal.js:1:465298)
```

**To Reproduce**
Run pyright on this repo: https://github.com/aerospike/aerospike-admin

**Expected behavior**
There should be no error.

**Screenshots or Code**
The error shows up in `asadm.py`.   The error appears to be unrelated to the code.  Highlighted on shebang at top of file.

![image](https://user-images.githubusercontent.com/22361806/109856937-ee0abe80-7c0e-11eb-89d7-8b89b665dc73.png)

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

v2021.2.4 as a VSCode extension.
pyright 1.1.117 as CLI

**Additional context**
Running on ubuntu 20

**Comment by erictraut** [comment]
Thanks for the great bug report and repro case. I've found and fixed the problem. It occurs when a function is declared in a local scope but is declared "global", like this:

```python
def func1():
   global func2
   def func2():
      pass
```

This is a very strange construct, but it is legal Python.

This will no longer cause a crash in the next release.

**Comment by jdogmcsteezy** [comment]
Thanks for the quick turnaround!  I assume you are talking about terminal.py?

**Comment by erictraut** [comment]
Yes, that's correct.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.118, which I just published. It will also be included in the next release of Pylance.

**Comment by jdogmcsteezy** [comment]
Works!
